public/app/features/alerting/unified/components/settings/AlertmanagerConfig.tsx

```diff
@@ -0,0 +1,185 @@
+import { useUnifiedAlertingSelector } from '../../hooks/useUnifiedAlertingSelector';
+import { GRAFANA_RULES_SOURCE_NAME, isVanillaPrometheusAlertManagerDataSource } from '../../utils/datasource';
+import { Spacer } from '../Spacer';
+
+export interface FormValues {
+  configJSON: string;
+}
+
+interface Props {
+  alertmanagerName: string;
+  onDismiss: () => void;
+  onSave: (dataSourceName: string, oldConfig: string, newConfig: string) => void;
+  onReset: (dataSourceName: string) => void;
+}
+
+export default function AlertmanagerConfig({ alertmanagerName, onDismiss, onSave, onReset }: Props): JSX.Element {
+  const { loading: isDeleting, error: deletingError } = useUnifiedAlertingSelector((state) => state.deleteAMConfig);
+  const { loading: isSaving, error: savingError } = useUnifiedAlertingSelector((state) => state.saveAMConfig);
+  const [showResetConfirmation, setShowResetConfirmation] = useState(false);
+
+  const readOnly = alertmanagerName ? isVanillaPrometheusAlertManagerDataSource(alertmanagerName) : false;
+  const styles = useStyles2(getStyles);
+
+  const {
+    currentData: config,
+    error: loadingError,
+    isSuccess: isLoadingSuccessful,
+    isLoading: isLoadingConfig,
+  } = useAlertmanagerConfig(alertmanagerName);
+
+  const defaultValues = {
+    configJSON: config ? JSON.stringify(config, null, 2) : '',
+  };
+
+  const {
+    register,
+    setValue,
+    setError,
+    handleSubmit,
+    formState: { errors },
+  } = useForm<FormValues>({
+    defaultValues,
+  });
+
+  useEffect(() => {
+    if (savingError) {
+      setError('configJSON', { type: 'deps', message: savingError.message });
+    }
+  }, [savingError, setError]);
+
+  useEffect(() => {
+    if (deletingError) {
+      setError('configJSON', { type: 'deps', message: deletingError.message });
+    }
+  }, [deletingError, setError]);
+
+  // manually register the config field with validation
+  register('configJSON', {
+    required: { value: true, message: 'Configuration cannot be empty' },
+    validate: (value: string) => {
+      try {
+        JSON.parse(value);
+        return true;
+      } catch (e) {
+        return e instanceof Error ? e.message : 'JSON is invalid';
+      }
+    },
+  });
+
+  const handleSave = handleSubmit(
+    (values: FormValues) => {
+      onSave(alertmanagerName, defaultValues.configJSON, values.configJSON);
+    },
+    (errors) => {
+      console.error(errors);
+    }
+  );
+
+  const isOperating = isLoadingConfig || isDeleting || isSaving;
+
+  /* loading error, if this fails don't bother rendering the form */
+  if (loadingError) {
+    return (
+      <Alert severity="error" title="Failed to load Alertmanager configuration">
+        {loadingError.message ?? 'An unkown error occurred.'}
+      </Alert>
+    );
+  }
+
+  /* resetting configuration state */
+  if (isDeleting) {
+    return (
+      <Alert severity="info" title="Resetting Alertmanager configuration">
+        It might take a while...
```

Comment:
Is it something that we have to poll for? Message could be more along the lines of "Please check back later"?

Reply:
I've taken this verbatim from our earlier implementation but agree that the messaging is a bit odd – we don't currently poll for this change.